src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/validator/ActionPlanSessionValidator.kt

```diff
@@ -4,41 +4,49 @@ import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.Code
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.FieldError
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config.ValidationError
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.AddressDTO
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.UpdateAppointmentDTO
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AppointmentDeliveryType
 
 @Component
 class ActionPlanSessionValidator {
-
+  companion object {
+    val postCodeRegex = Regex("^[A-Z]{1,2}\\d[A-Z\\d]? ?\\d[A-Z]{2}\$")
```

Comment:
starting out of my habits when entering my postcode, I don't usually bother with uppercasing it, or I may double-space it when I'm on mobile, which would fail here I think -- do we normalise this somewhere?

I think it should be normalised here, rather than at client side (i.e. remove all spaces + uppercase)